### PR TITLE
[RFC] fix #270, support `appendTsSuffixTo` config

### LINKED ISSUE #270: Possibility to support vue-loader?
Thanks for building ts-loader for webpack! I wonder whether ts-loader can support vue.js' single file component.

Related issue in vue-loader
https://github.com/vuejs/vue-loader/issues/109

The current problem is ts-loader cannot resolve files with extension name `.vue`. However, TypeScript compiler supports [customizing module resolution](https://github.com/Microsoft/TypeScript/wiki/Using-the-Compiler-API#customizing-module-resolution).

Since ts-loader has already implemented a custom `resolveModuleNames` function, supporting it is feasible.

 I tried locally and it works. However I'm a newbie to webpack and I think my try is dirty and hacky...

Do you have any good ideas? I'm happy to implement them.

### PR DESCRIPTION
**Pull request by HerringtonDarkholme**
~Tests need more handling. Vue loader will produces absolute file path that is path dependent.
And it even produce hash that is machine specific! How can I handle these cases?~

Added _flaky mark to vue's test

### CODE REVIEW

**Comment by johnnyreilly** [comment]
> Vue loader will produces absolute file path that is path dependent.
> And it even produce hash that is machine specific!

Don't worry; our test pack already caters for [this](https://github.com/TypeStrong/ts-loader/blob/master/test/comparison-tests/create-and-execute-test.js#L179)

```
/**
 * replace the hash if found in the output since it can change depending
 * on environments and we're not super interested in it and normalize
 * line endings
 **/
function cleanHashFromOutput(stats, webpackOutput) {
    if (stats) {
        glob.sync('**/*', { cwd: webpackOutput, nodir: true }).forEach(function (file) {
            var content = fs.readFileSync(path.join(webpackOutput, file), 'utf-8')
                .split(stats.hash).join('[hash]')
                .split('\r\n').join('\n');
            fs.writeFileSync(path.join(webpackOutput, file), content);
        });
    }
}
```

**Comment by johnnyreilly** [comment]
Is this ready for me to take a look at?  I've been holding off until you give me the go-ahead.

**Comment by HerringtonDarkholme** [comment]
Yes, I have had _flaky_ mark to vue's test. I think as long as the test does not exit with error. It is ok.

File: test/comparison-tests/vue/expectedOutput-2.0/bundle.transpiled.js [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+	__vue_options__.staticRenderFns = __vue_template__.staticRenderFns
+
+	/* hot reload */
+	if (false) {(function () {
+	  var hotAPI = require("vue-hot-reload-api")
+	  hotAPI.install(require("vue"), false)
+	  if (!hotAPI.compatible) return
+	  module.hot.accept()
+	  if (!module.hot.data) {
+	    hotAPI.createRecord("data-v-39ef11c7", __vue_options__)
```
Comment (HerringtonDarkholme) [inline test/comparison-tests/vue/expectedOutput-2.0/bundle.transpiled.js]: Here, the hash code is even emitted in bundle.js

**Comment by johnnyreilly** [comment]
Ah - in that case we can probably amend the test pack to cater for hashes in  the bundle output as well.  Actually this is kind of handy as this is the first example of a hot module reloading test in the test pack that I'm aware of.  But I can take a look at that later

**Comment by johnnyreilly** [comment]
Looks good to me 👍

**Comment by HerringtonDarkholme** [comment]
Thanks!

**Comment by johnnyreilly** [comment]
My pleasure; I've invited you to join TypeStrong so you'd have commit rights on ts-loader.  

In order that we can release I think we just need to do the following:
- update the docs to include details about the new option.  Probably worth providing a quick example of how to use this with vuejs for other people to find
- follow the standard instructions [here](https://github.com/TypeStrong/ts-loader/blob/master/CONTRIBUTING.md#publishing)

**Comment by johnnyreilly** [comment]
Do you fancy handling that?  I'd propose that this should become ts-loader v1.1.0

**Comment by HerringtonDarkholme** [comment]
I think I would provide the doc part. But releasing version goes to you 😉

**Comment by johnnyreilly** [comment]
That's fine 😄 - let me know when the docs are updated and I'll handle it

**Pull request merged**